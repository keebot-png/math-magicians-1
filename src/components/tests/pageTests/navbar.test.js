import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../../navbarHeading/navbar';

describe('Testing the links in the navbar', () => {
  test('Testing to show that all links are displaying correctly', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );

    expect(screen.getAllByText('Home')).toMatchSnapshot();
    expect(screen.getAllByText('Calculator')).toMatchSnapshot();
    expect(screen.getAllByText('Quote')).toMatchSnapshot();
  });

  test('Test for navigating to Home Page', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );

    fireEvent.click(screen.getByText('Home'));

    expect(screen.getByText('Home')).toMatchInlineSnapshot(`
<a
  href="/"
>
  Home
</a>
`);
  });

  test('Test for navigating to Calculator page', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );

    fireEvent.click(screen.getByText('Calculator'));

    expect(screen.getByText('Calculator')).toMatchInlineSnapshot(`
<a
  href="/calculator"
>
  Calculator
</a>
`);
  });

  test('Test for navigating to the Quote page', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );

    fireEvent.click(screen.getByText('Quote'));

    expect(screen.getByText('Quote')).toMatchInlineSnapshot(`
<a
  href="/quote"
>
  Quote
</a>
`);
  });
});
