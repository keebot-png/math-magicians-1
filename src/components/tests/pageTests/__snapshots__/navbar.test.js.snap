// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the links in the navbar Testing to show that all links are displaying correctly 1`] = `
Array [
  <a
    href="/"
  >
    Home
  </a>,
]
`;

exports[`Testing the links in the navbar Testing to show that all links are displaying correctly 2`] = `
Array [
  <a
    href="/calculator"
  >
    Calculator
  </a>,
]
`;

exports[`Testing the links in the navbar Testing to show that all links are displaying correctly 3`] = `
Array [
  <a
    href="/quote"
  >
    Quote
  </a>,
]
`;
